### DISCUSSION TITLE
Underline TextNode based on cursor position

### DISCUSSION THREAD

**Discussion by tomaszferens** [facebook/lexical#4698] [Q&A]
I want to achieve a following functionality:

![image](https://github.com/facebook/lexical/assets/17815179/eb0e7c8c-0b85-4492-bd08-b96a16cfeaba)

(notice that "are" is <ins>underlined</ins> while other words are normal). The cursor position decides what TextNode should be underlined.

I tried to use `editor.registerNodeTransform(TextNode, (textNode) => {})` to split single TextNode into multiple TextNodes (splitting by space " " character) and `editor.registerCommand(SELECTION_CHANGE_COMMAND, () => {})` to achieve it but i'm having issues with lexical infinitely updating itself.

Here is a gist:

```jsx
import { useLexicalComposerContext } from "@lexical/react/LexicalComposerContext";
import { useEffect } from "react";
import { TextNode } from "lexical";
import {
  $getSelection,
  SELECTION_CHANGE_COMMAND,
  COMMAND_PRIORITY_NORMAL
} from "lexical";

export default function UnderlinePlugin() {
  const [editor] = useLexicalComposerContext();

  useEffect(() => {
    editor.registerNodeTransform(TextNode, (textNode) => {
      const textContent = textNode.getTextContent();

      const whiteSpaceIndices = [];
      for (let i = 0; i < textContent.length; i++) {
        if (textContent[i] === " ") {
          whiteSpaceIndices.push(i);
          whiteSpaceIndices.push(i + 1);
        }
      }

      editor.update(() => {
        textNode.splitText(...whiteSpaceIndices);
      });
    });
  }, []);

  useEffect(() => {
    return editor.registerCommand(
      SELECTION_CHANGE_COMMAND,
      () => {
        const selection = $getSelection();

        const [node] = selection.getNodes();

        // node.setFormat("underline"); // <- causes lexical to indefinitely update
      },
      COMMAND_PRIORITY_NORMAL
    );
  }, []);

  return null;
}
```

[DEMO](https://codesandbox.io/s/objective-boyd-p5nr7n?file=/src/plugins/UnderlinePlugin.js:277-292)

what am I doing wrong? 🤔

**Comment by criloz** [comment]
@tomaszferens did you get the answer to handle this kind of node transformations?
